### ISSUE TITLE
Typography controls can't be disabled on block level

### ISSUE THREAD

**Issue by lunule** [WordPress/gutenberg#59722]
### Description

Typography controls configuration in `theme.json` doesn't work properly.

The example I tested is the `fontSizes` setting, with the following scenario:

- I have font sizes configured on global level (in the `settings` section)
- most of the blocks should have typography controls, but not the `core/heading` block where I want fix font sizes
   - that can't be changed by the user
   - that are not available in the array of font sizes mentioned above
- as a result, I need to disable the font size selector for the `core/heading` block only - but this is currently impossible.

I tested the following variations:

- `"fontSize: false"`
- `"fontSizes: false"`
- `"fontSize: []"`
- `"fontSizes: []"`

... in all possible combos, and by applying these combos 
- in the `styles` section for the `core/heading` block
- in the `elements` section for the `h1` element

None of these variations work, it seems that, currently, it's impossible to disable the font size selector on block level if/when the `fontSizes` array in the `settings` section is not an empty array (or if it's not set to false). 

### Step-by-step reproduction instructions

The issue can be reproduced easily by following the above-mentioned test-scenario.

In a nutshell:
1. Configure font sizes for the `settings` section's `fontSizes` array
2. then try to disable the font selector specifically for the `core/heading` block.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by thesandybridge** [comment]
This isn't a bug, it is just a missing feature. You can disable custom font sizes using `"customFontSize": false` but if you want to specifically disable the font size selector altogether for a specific block you will need to create a higher order component and essentially reconstruct the InspectorControls for that block. I am not aware of a way to do this with filters, but there could be. I am just trying to point you in the right direction.

**Comment by lunule** [comment]
Hey @thesandybridge - thanks for your feedback and for your advice (unfortunately, I'm not skilled enough in React.js to handle this process) - I'm reposting this as a feature request.
